### PR TITLE
chore (react-router-native): remove redundant types

### PR DESCRIPTION
**Pull request by 43081j**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).
- [x] If a package was never on Definitely Typed, you don't need to do anything. (If you wrote a package and provided types, you don't need to register it with us.)
- [x] Delete the package's directory.
- [x] Add it to `notNeededPackages.json`.

---

it ships its own types since 6.4.0:
https://unpkg.com/browse/react-router-native@6.4.0/dist/index.d.ts

### CODE REVIEW

**Comment by 43081j** [comment]
will trigger a new build once microsoft/DefinitelyTyped-tools#622 is merged

File: types/react-router-navigation/react-router-navigation-tests.tsx
Comment (43081j) [inline types/react-router-navigation/react-router-navigation-tests.tsx:3]: this is a tests-only dependency, which DT maintainers have said is bad practice in the past.

rather than introducing a production dependency to make this work (one the actual types don't use), i've added some mock interfaces/functions since the shape doesn't actually matter for testing this as long as the parent accepts children.

**Comment by DangerBotOSS** [comment]
Inspecting the JavaScript source for this package found some properties that are not in the .d.ts files.
The check for missing properties isn't always right, so take this list as advice, not a requirement.
## react-router-navigation ([<kbd>unpkg</kbd>](https://unpkg.com/browse/react-router-navigation@latest/))
was missing the following properties:
1. DefaultNavigationRenderer
1. DefaultTabsRenderer
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 88322a8dc8f512c9971d905a2c902250256a1293
</p>

**Review by kaoDev — APPROVED** [review]

File: types/react-router-navigation/react-router-navigation-tests.tsx
Comment (kaoDev) [reply types/react-router-navigation/react-router-navigation-tests.tsx:3]: thanks for cleaning up

**Review by jakebailey — APPROVED** [review]

**Comment by 43081j** [comment]
ready to merge 🍺

**Pull request merged**
